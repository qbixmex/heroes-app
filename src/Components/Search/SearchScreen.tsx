const SearchScreen = () => {
  return (
    <div>
      <h1 className="text-center display-1 green">DC Screen</h1>
    </div>
  );
};

export default SearchScreen;
