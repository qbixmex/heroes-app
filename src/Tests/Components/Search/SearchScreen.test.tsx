import { mount } from 'enzyme';
import SearchScreen from '../../../Components/Search/SearchScreen';
import { AuthContext } from '../../../Auth/authContext';
import { MemoryRouter } from 'react-router-dom';

// @ts-ignore
describe("Tests on SearchScreen", () => {
  test("Should match with snapshot with default values", () => {

    const wrapper = mount(
      <MemoryRouter initialEntries={ ["/search"] }>
        <SearchScreen />
      </MemoryRouter>
    );

    const title = wrapper.find("#searchTitle").text().trim();
    const alert = wrapper.find(".alert").text().trim();

    expect( wrapper ).toMatchSnapshot();
    expect( title ).toBe("Search Screen");
    expect( alert ).toBe("Type your favorite hero above");

  });

});
