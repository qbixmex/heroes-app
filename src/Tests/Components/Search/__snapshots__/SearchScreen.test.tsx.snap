// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on SearchScreen Should match with snapshot with default values 1`] = `
<div>
  <h1
    className="text-center display-1 green"
    id="searchTitle"
  >
    Search Screen
  </h1>
  <hr />
  <div
    className="animate__animated animate__fadeIn"
  >
    <form
      onSubmit={[Function]}
    >
      <div
        className="row mb-5"
      >
        <div
          className="col-12 col-md-10 mb-3 mb-md-0"
        >
          <input
            autoComplete="off"
            className="form-control"
            id="search"
            name="searchText"
            onChange={[Function]}
            placeholder="Search a hero ..."
            type="text"
            value=""
          />
        </div>
        <div
          className="col-12 col-md-2"
        >
          <div
            className="d-grid"
          >
            <button
              className="btn btn-outline-primary"
              type="submit"
            >
              Search
            </button>
          </div>
        </div>
      </div>
    </form>
    <div
      className="row"
    >
      <div
        className="alert alert-info text-center mt-3"
      >
        <b>
          Type your favorite hero above
        </b>
      </div>
    </div>
  </div>
</div>
`;
